### DISCUSSION TITLE
Remove unused atoms from memory?

### DISCUSSION THREAD

**Discussion by rafark** [pmndrs/jotai#2588] [Q&A] [answered]
Sorry I have no clue about this.

Say I'm creating an atom for a new item and i have an object that maps ids to atoms:

```
const firstItem = {
    id: 10, 
    name: 'my Item'
}

const items = {
    10: atom('the atom'),
    20: atom('the atom'),
    ...etc
}

```
If somewhere in my app i delete the item with the id of  10, i want to get rid of its atom too to prevent a memory leak. How? My app will potentially have thousands of atoms and just imagine the memory leaks if i don't remove these atoms when their associated item is removed.

**Comment by dai-shi** [comment, marked as answer]
`delete items['10']`.

Think of Jotai store being just a WeakMap (and it actually is, except for tracking "mounted" atoms).

**Reply by rafark** [reply]
Thank for taking the time to answer!
